harness/determined/common/storage/base.py

```diff
@@ -2,11 +2,17 @@
 import contextlib
 import os
 import pathlib
-from typing import Any, Callable, Dict, Iterator, Optional, Union
+from typing import Any, Callable, Dict, Iterator, List, Optional, Union
+
+# Paths should be a sorted list of paths relative to the checkpoint root that indicate what paths
+# should be uploaded.  A directory should always appear in Paths if any subpath under that directory
+# appears in Paths.
+Paths = List[str]
+
 
 # Selector accepts a path relative to the checkpoint root, and returns a boolean indicating if the
-# path should be downloaded. For every path selected, all parent directories are also selected
-# (even if the selector returns False for them).
+# path should be uploaded/downloaded. For every path selected, all parent directories are also
```

Comment:
but... upload doesn't appear to take a selector?

Reply:
yeah, my bad... this upload takes paths, upload in checkpoint context takes selectors. fixed.